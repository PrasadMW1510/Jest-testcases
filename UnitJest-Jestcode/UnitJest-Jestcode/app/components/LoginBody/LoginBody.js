/**
 *
 * LoginBody
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import './LoginBody.scss';

function LoginBody(props) {
  return <div className="login-body">{props.children}</div>;
}

LoginBody.propTypes = {
  children: PropTypes.node,
};

export default LoginBody;
