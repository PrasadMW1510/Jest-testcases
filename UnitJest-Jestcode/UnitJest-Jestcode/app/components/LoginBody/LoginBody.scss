.login-body {
  height: 679px;
  width: 1019px;
  margin: 0 auto;
  background: url('~images/SAM-login.png');
  background-size: 100%;
  position: relative;
}
