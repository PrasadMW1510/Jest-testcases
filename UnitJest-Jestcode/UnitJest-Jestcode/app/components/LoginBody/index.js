/**
 *
 * LoginBody
 *
 */

export { default } from './LoginBody';
