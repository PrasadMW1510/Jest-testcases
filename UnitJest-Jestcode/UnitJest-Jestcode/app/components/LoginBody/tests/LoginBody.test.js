import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import LoginBody from '../index';

describe('<LoginBody />', () => {
  let wrapper = null;

  beforeEach(() => {
    wrapper = shallow(<LoginBody />);
  });

  it('Expect to render itself', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Expect to render children', () => {
    wrapper = shallow(<LoginBody> hello </LoginBody>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
