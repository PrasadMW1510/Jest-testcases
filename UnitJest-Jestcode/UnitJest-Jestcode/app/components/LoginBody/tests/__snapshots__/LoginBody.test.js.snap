// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginBody /> Expect to render children 1`] = `
<div
  className="login-body"
>
   hello 
</div>
`;

exports[`<LoginBody /> Expect to render itself 1`] = `
<div
  className="login-body"
/>
`;
